heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/WebSink.java

```diff
@@ -192,8 +195,20 @@ public void processRecord(MetricsRecord record) {
     if (isFlatMetrics) {
       metricsCache.putAll(processMetrics(source + "/", record.getMetrics()));
     } else {
-      Map<String, Double> sourceMap = processMetrics("", record.getMetrics());
-      metricsCache.put(source, sourceMap);
+      Map<String, Double> sourceCache;
+      Object sourceObj = metricsCache.getIfPresent(source);
+      if (sourceObj instanceof Map) {
+        @SuppressWarnings("unchecked")
+        Map<String, Double> castObj =  (Map<String, Double>) sourceObj;
+        sourceCache = castObj;
```

Comment:
Can we merge L202 and L203 into one line:
sourceCache = (Map<String, Double>) sourceObj;